### PR TITLE
WITI-319-Add dropdown fields

### PR DESCRIPTION
**Pull request by RamanSel**
#### What problem is this solving?

Adding a dropdown builder and useOnRegistration boolean for Custom fields in order to have them appear on organization registration page

#### How to test it?

Custom fields builder:
https://witi319--whitebird.myvtex.com/admin/b2b-organizations/organizations/#/custom-fields

Custom fields saved after registration: 
https://witi319--whitebird.myvtex.com/admin/b2b-organizations/organizations/f68bac65-5c68-11ed-83ab-0a1be9fca741/#/default

#### Screenshots or example usage:

image

Screen Shot 2022-11-04 at 14 24 13

Screen Shot 2022-11-04 at 14 24 25

#### Describe alternatives you've considered, if any.

Considered hard-coding the dropdown, but that would not work as the app is used both for renwil and whitebird.

#### Related to / Depends on

https://github.com/syatt-io/b2b-organizations-graphql/pull/14

### CODE REVIEW

File: react/admin/DefaultCustomField.tsx [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+      }
+
+      handleUpdate(index, updatedCustomField)
+
+      return newDropdownValues
+    })
+  }
+
+  const handleAddDropdownItem = () => {
+    setDropdownValuesLocal((prev: any) => {
```
Comment (jonmast) [inline react/admin/DefaultCustomField.tsx:99]: `prev` should be of type `DropdownValue[]`, not `any`

File: react/admin/DefaultCustomField.tsx [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+          icon={<IconPlus title="Add field" />}
+          aria-label="Add field"
+          onClick={handleAddDropdownItem}
+          csx={{
+            marginTop: '28px',
+            marginLeft: '10px',
+          }}
+        />
+      ) : null}
+      {/* // TODO  */}
```
Comment (jonmast) [inline react/admin/DefaultCustomField.tsx:222]: What is this for?

File: react/admin/DefaultCustomField.tsx [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+}) => {
+  const [dropdownValuesLocal, setDropdownValuesLocal] = useState<
+    DropdownValue[]
+  >([])
+
+  const [useOnRegistrationLocal, setUseOnRegistrationLocal] = useState(false)
+
+  useEffect(() => {
+    setDropdownValuesLocal(dropdownValues as DropdownValue[])
+    setUseOnRegistrationLocal(!!useOnRegistration as boolean)
```
[the remark this answers is not in the document]
Comment (RamanSel) [reply react/admin/DefaultCustomField.tsx]: ✅ removed

File: react/admin/DefaultCustomField.tsx [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+interface CustomFieldProps {
+  index: number
+  customField: CustomField
+  name: string
+  handleUpdate: (index: number, customField: CustomField) => void
+}
+
+interface DropdownValue {
+  value: 'string'
+  label: 'string'
```
[the remark this answers is not in the document]
Comment (RamanSel) [reply react/admin/DefaultCustomField.tsx]: Ah, that's why it wasn't working correctly 🤦, thank you

**Review by jonmast — APPROVED** [review]

**Pull request merged**